jax/_src/numpy/fft.py

```diff
@@ -55,7 +58,20 @@ def _fft_core(func_name, fft_type, a, s, axes, norm):
     axes = tuple(range(a.ndim - len(axes), a.ndim))
     a = jnp.moveaxis(a, orig_axes, axes)
 
-  if s is None:
+  if s is not None:
+    a = jnp.asarray(a)
+    in_s = list(a.shape)
+    for axis, x in zip(axes, s):
+      in_s[axis] = x
+    if fft_type == xla_client.FftType.IRFFT:
+      in_s[-1] = (in_s[-1] // 2 + 1)
+
+    if any([x != y for x, y in zip(in_s, a.shape)]):
+      # Cropping
+      a = a[tuple(map(slice, in_s))]
+      # Padding
+      a = lax.pad(a, lax._const(a, 0), [(0, x-y, 0) for x, y in zip(in_s, a.shape)])
```

Comment:
I would suggest using the higher level `jnp.pad` here:
```suggestion
      a = jnp.pad(a, [(0, x-y) for x, y in zip(in_s, a.shape)])
```